### PR TITLE
Add the Checkstyle localization

### PR DESCRIPTION
**Pull request by dev-yakuza**
I added the documentation about how to add and use the `Checkstyle` localization.

- Related: https://github.com/sider/runners/issues/1658

### CODE REVIEW

**Comment by dev-yakuza** [comment]
you can see the document on [here](https://deploy-preview-530--sider-docs.netlify.app/tools/java/checkstyle)

Screen Shot 2020-12-02 at 9 21 48

**Comment by ybiquitous** [comment]
> * Fix: [sider/runners#1658](https://github.com/sider/runners/issues/1658)

Is it correct?

**Comment by dev-yakuza** [comment]
Sorry, I will change it 🙇‍♂️

File: docs/tools/java/checkstyle.md [hunk 1]
```diff
@@ -82,3 +82,28 @@ By default, nothing will be ignored.
...
+If you want to use the localization on Sider, you need to do as follows.
+
+1. Download(or create) a Checkstyle configuration file(e.g. google_checks.xml).
+1. Add the `localeCountry` and `localeLanguage` properties to the configuration file like below. (See the details on [Checkstyle documentation](https://checkstyle.sourceforge.io/config.html#Checker_Examples))
+
+   ```xml
+   <module name="Checker">
+     <property name="localeCountry" value="DE"/>
+     <property name="localeLanguage" value="de"/>
+     ...
```
Comment (ybiquitous) [inline docs/tools/java/checkstyle.md:99]: [suggestion]

- For many users in Japan for Sider.
- Change to valid XML.

```suggestion
     <property name="localeCountry" value="JP"/>
     <property name="localeLanguage" value="ja"/>
     <!-- ... -->
```

File: docs/tools/java/checkstyle.md [hunk 2]
```diff
@@ -82,3 +82,28 @@ By default, nothing will be ignored.
...
+   </module>
+   ```
+
+1. Modify(or create) the `sider.yml` like below.
+
+   ```yml
+   linter:
+     checkstyle:
+       config: [Your Rule File Name].xml
```
Comment (ybiquitous) [inline docs/tools/java/checkstyle.md:108]: [suggestion] Use the example file name above:

```suggestion
       config: your_checkstyle_ruleset.xml
```

File: docs/tools/java/checkstyle.md [hunk 3]
```diff
@@ -82,3 +82,28 @@ By default, nothing will be ignored.
 ### `properties`
 
 This option allows you to specify the properties file passed to `checkstyle`. The value will be passed to Checkstyle's `-p` option.
+
+## Localization
+
+The `Checkstyle` supports localization output messages. You can see the details on [Checkstyle documentation](https://checkstyle.org/config_system_properties.html#Localisation_Support). Also, You can see the supported languages in [Checkstyle GitHub](https://github.com/checkstyle/checkstyle/tree/432bafd49ed9d801f44a04ad710cc9372538e588/src/main/resources/com/puppycrawl/tools/checkstyle/checks/sizes).
```
[the remark this answers is not in the document]
Comment (dev-yakuza) [reply docs/tools/java/checkstyle.md]: Fixed https://github.com/sider/sider-docs/pull/530/commits/cdf4e1e5cdb858f02a8f2c903f583a629a53e076

File: docs/tools/java/checkstyle.md [hunk 4]
```diff
@@ -82,3 +82,28 @@ By default, nothing will be ignored.
...
+
+If you want to use the localization on Sider, you need to do as follows.
+
+1. Download(or create) a Checkstyle configuration file(e.g. google_checks.xml).
```
[the remark this answers is not in the document]
Comment (dev-yakuza) [reply docs/tools/java/checkstyle.md]: Fixed https://github.com/sider/sider-docs/pull/530/commits/cdf4e1e5cdb858f02a8f2c903f583a629a53e076

File: docs/tools/java/checkstyle.md [hunk 5]
```diff
@@ -82,3 +82,28 @@ By default, nothing will be ignored.
...
+1. Add the `localeCountry` and `localeLanguage` properties to the configuration file like below. (See the details on [Checkstyle documentation](https://checkstyle.sourceforge.io/config.html#Checker_Examples))
```
[the remark this answers is not in the document]
Comment (dev-yakuza) [reply docs/tools/java/checkstyle.md]: Fixed https://github.com/sider/sider-docs/pull/530/commits/cdf4e1e5cdb858f02a8f2c903f583a629a53e076

File: docs/tools/java/checkstyle.md [hunk 6]
```diff
@@ -82,3 +82,28 @@ By default, nothing will be ignored.
...
+
+   ```xml
+   <module name="Checker">
+     <property name="localeCountry" value="DE"/>
+     <property name="localeLanguage" value="de"/>
+     ...
+   </module>
+   ```
+
+1. Modify(or create) the `sider.yml` like below.
```
[the remark this answers is not in the document]
Comment (dev-yakuza) [reply docs/tools/java/checkstyle.md]: Fixed https://github.com/sider/sider-docs/pull/530/commits/cdf4e1e5cdb858f02a8f2c903f583a629a53e076

File: docs/tools/java/checkstyle.md (hunk 1 above)
Comment (dev-yakuza) [reply docs/tools/java/checkstyle.md]: Fixed https://github.com/sider/sider-docs/pull/530/commits/cdf4e1e5cdb858f02a8f2c903f583a629a53e076

File: docs/tools/java/checkstyle.md (hunk 2 above)
Comment (dev-yakuza) [reply docs/tools/java/checkstyle.md]: Fixed https://github.com/sider/sider-docs/pull/530/commits/cdf4e1e5cdb858f02a8f2c903f583a629a53e076

File: docs/tools/java/checkstyle.md [hunk 7]
```diff
@@ -82,3 +82,28 @@ By default, nothing will be ignored.
 ### `properties`
 
 This option allows you to specify the properties file passed to `checkstyle`. The value will be passed to Checkstyle's `-p` option.
+
+## Localization
+
+`Checkstyle` supports the localization of output messages. You can see the details on [Checkstyle documentation](https://checkstyle.org/config_system_properties.html#Localisation_Support). Also, you can see the supported languages in [Checkstyle's GitHub](https://github.com/checkstyle/checkstyle/tree/432bafd49ed9d801f44a04ad710cc9372538e588/src/main/resources/com/puppycrawl/tools/checkstyle/checks/sizes).
```
Comment (ybiquitous) [inline docs/tools/java/checkstyle.md:88]: Not need a code style for "Checkstyle":

```suggestion
Checkstyle supports the localization of output messages. You can see the details on [Checkstyle documentation](https://checkstyle.org/config_system_properties.html#Localisation_Support). Also, you can see the supported languages in [Checkstyle's GitHub](https://github.com/checkstyle/checkstyle/tree/432bafd49ed9d801f44a04ad710cc9372538e588/src/main/resources/com/puppycrawl/tools/checkstyle/checks/sizes).
```

Comment (dev-yakuza) [reply docs/tools/java/checkstyle.md:88]: Thanks!

**Review by ybiquitous — APPROVED** [review]
LGTM 👍

**Pull request merged**